UI MoneyRequestHeader Polish

s77rt opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
1. Set flex 1 to Views so the workspace name does not overflow.
2. Adjusted subscript avatar position, background color and border radius (more info in [comment](https://github.com/Expensify/App/pull/18775#issuecomment-1546117364)).
3. Adjusted MoneyRequestHeader bottom padding to meet the design mockup (20px).
4. The `<ReportActionItemCreated />` (welcome message, avatars and bg art) will be rendered only if the report is a chat report.
5. Reduced section titles heights

### Fixed Issues
$ https://github.com/Expensify/App/issues/18773


### Tests
Same as https://github.com/Expensify/App/pull/18537 but request money from a workspace/user with a very long name
1. Verify name does not overflow and shows ellipsis instead
2. Verify there is no welcome message and that the background is blank

![web](https://github.com/Expensify/App/assets/16493223/65f5088d-8bb1-4248-8786-44a7b2ebdc8a)


- [ ] Verify that no errors appear in the JS console

### Offline tests
n/a

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
3. Upload an image via copy paste
4. Verify a modal appears displaying a preview of that image
--->

same as Tests ^

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->
![web](https://github.com/Expensify/App/assets/16493223/65f5088d-8bb1-4248-8786-44a7b2ebdc8a)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

s77rt committed:
Truncate long names

s77rt committed:
fix avatar background, position and border radius

s77rt commented on src/components/AvatarWithDisplayName.js:
```diff
@@ -53,9 +54,10 @@ const AvatarWithDisplayName = (props) => {
+                <View style={[styles.flex1, styles.flexRow, styles.alignItemsCenter, styles.justifyContentBetween]}>
                     {isExpenseReport ? (
                         <SubscriptAvatar
+                            backgroundColor={themeColors.highlightBG}
```
This is taken from `MoneyRequestHeader`

s77rt committed:
fix MoneyRequestHeader padding

Julesssss reviewed (dismissed):
Looking good. I'm going to cc Shawn for a final design review.

Julesssss commented:
I kicked off an automated builds run [here](https://github.com/Expensify/App/actions/runs/4956770413). Once complete, I'll share the links with Shawn for review.

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18775/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18775/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/18775/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/18775/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18775/NewExpensify.dmg  | https://18775.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/18775/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://18775.pr-testing.expensify.com)  |

s77rt commented on src/styles/StyleUtils.js:
```diff
@@ -37,7 +37,7 @@ const avatarBorderSizes = {
     [CONST.AVATAR_SIZE.SMALL]: variables.componentBorderRadiusMedium,
     [CONST.AVATAR_SIZE.HEADER]: variables.componentBorderRadiusMedium,
-    [CONST.AVATAR_SIZE.DEFAULT]: variables.componentBorderRadiusNormal,
+    [CONST.AVATAR_SIZE.DEFAULT]: variables.componentBorderRadiusMedium,
```
This will effect workspace avatars.
![Screenshot from 2023-05-12 16-48-30](https://github.com/Expensify/App/assets/16493223/600a8ab3-fb9d-448f-a7b5-b5d05e5a1d16)
![Screenshot from 2023-05-12 16-48-23](https://github.com/Expensify/App/assets/16493223/3f0a4c2e-f5a2-4b28-9630-236a9acadeee)

cc @shawnborton

shawnborton replied:
Hmm we should only be changing the border radius at the super small size when it is overlaid on top of a user avatar.

s77rt replied:
Updated.

s77rt committed:
Merge branch 'Expensify:main' into iou-header-polish

s77rt committed:
Removed redundant border radius in Avatar

s77rt commented:
**Key notes:**

1. Added `iconAdditionalStyles` prop to Avatar to pass styles and avoid the double border radius. Previously there was 3 border radius values (20px for default avatar, 4px inner, 6px outer). Now we only have one which is 6px.
2. We can now pass `imageStyles=null` to clear the preset styles. This is needed to remove the 20px border radius and unset the unnecessary width and height for the subscript avatar.
3. `StyleUtils.getAvatarBorderWidth` will now return an `object` instead of a `number`.
4. Removed the extra `View` in `SubscriptAvatar`. Will be using the existing `containerStyles` prop.

I think the rest is self-explanatory.

___

Here are the results:
![Screenshot from 2023-05-12 19-15-52](https://github.com/Expensify/App/assets/16493223/1e319426-75d8-4eb4-8185-8a347bd3e59f)
![Screenshot from 2023-05-12 19-03-14](https://github.com/Expensify/App/assets/16493223/0de45ff3-ce22-40ad-ba9d-eaf65d15acaf)

cc @shawnborton

s77rt commented on src/styles/StyleUtils.js:
```diff
@@ -107,12 +107,12 @@ function getAvatarExtraFontSizeStyle(size) {
     const AVATAR_SIZES = {
         [CONST.AVATAR_SIZE.DEFAULT]: 3,
-        [CONST.AVATAR_SIZE.SMALL_SUBSCRIPT]: 2,
+        [CONST.AVATAR_SIZE.SMALL_SUBSCRIPT]: 1,
```
This was based on the removed border width from `secondAvatarSubscriptCompact`.

s77rt committed:
because consistency matters

shawnborton commented:
Looks good to me!

s77rt commented:
@0xmiroslav Yes. There are still 2 other enhancements but we may move those to a follow up PR. You can review this one as is.

s77rt commented:
Won't affect the current code:
- Reduce space of section titles.
- Match StatusBar color

You can refer to those on the linked issue.

mountiny reviewed (commented):
@s77rt Great job so far!

I have included one more point to the PR body:
> 4. Any thread, iou and expense report should not have the created/welcome message/ avatars or bg design. It should have blank bg

Lets remove this logic which stems from CREATED report action for threads and iou/expense reports

s77rt committed:
Merge branch 'Expensify:main' into iou-header-polish

s77rt committed:
render ReportActionItemCreated only if report is a chat report

s77rt commented:
@mountiny Updated. But I don't see why we create those "created"-reportActions in the first place. Do we actually need them for something else?

mountiny commented:
> But I don't see why we create those "created"-reportActions in the first place. Do we actually need them for something else?

yep its consistent with any report, it has a created report action which stores timestamp of when the report was created

s77rt commented:
Update regarding:
> Won't affect the current code:
> 
> * Reduce space of section titles.
> * Match StatusBar color
> 
> You can refer to those on the linked issue.

Reducing section titles height: waiting for @shawnborton confirmation on https://expensify.slack.com/archives/C02NK2DQWUX/p1683999682465889?thread_ts=1683808646.650509&cid=C02NK2DQWUX

Matching `StatusBar` color: I still can't find a smooth way to handle this. I'm trying to match the Status Bar color for the drawer and the report passively (control the color from one place) and currently having the idea of doing this based on current route.
- This is not smooth because the color may get changed before the MoneyRequestHeader is rendered (since we depend on the route and not on the component. If we depend on the component we will be controlling the color actively (and from different places) which may be hard to manage)
- The status bar color for the drawer is not consist. This is because I can't access the drawer state from the outside (outside the drawer). I'm not focusing on this much as the logic will probably change after navigation reboot.
-  This will not work on iOS. It does not support [setBackgroundColor](https://reactnative.dev/docs/statusbar#setbackgroundcolor-android).

Let me know if you have any ideas on how we can make this smoother or on how we should move in general.

https://github.com/Expensify/App/assets/16493223/e8d2d95b-82d6-4fe5-8c4e-78ac0affb54f

s77rt commented:
@mountiny No, it won't. Currently we set the status bar color only once on app load.
https://github.com/Expensify/App/blob/6c10bb2b36069b74e2dfcc5a97c42c2be6e767e1/src/components/CustomStatusBar/index.android.js#L10-L13

So on Android you will always see the same color (`appBG`).

mountiny commented:
ok, thanks! still not a blocker imho

mountiny reviewed (dismissed):
Looks and tests well for me but only done web

@0xmiroslav will you be able to run though the checklist on monday morning please?

s77rt committed:
Fixed Avatar size

s77rt commented:
Updated! cc @0xmiroslav

Julesssss commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -36,12 +36,12 @@ const defaultProps = {
-    const icons = ReportUtils.getIcons(props.report, props.personalDetails);
-
-    if (ReportUtils.isMoneyRequestReport(props.report.reportID) || ReportUtils.isTaskReport(props.report)) {
+    if (!ReportUtils.isChatReport(props.report)) {
```
I think we might still want to check for moneyRequest report, because users can request money from a workspace chat, not just 1:1 chat reports.

s77rt replied:
I'm not sure I see the point here. Workspace chats are of type `chat`. So the welcome message would still appear.

s77rt replied:
@0xmiroslav This is regarding the 4th point mentioned in PR details
> The <ReportActionItemCreated /> (welcome message, avatars and bg art) will be rendered only if the report is a chat report.

We want to show the welcome message only if the report is a chat report.

| Before | After |
|--------|-------|
|   <img src="https://github-production-user-asset-6210df.s3.amazonaws.com/16493223/237837074-0a6431d2-2850-4745-870d-8754daa5e832.png" />     |  <img src="https://github-production-user-asset-6210df.s3.amazonaws.com/16493223/238158065-65f5088d-8bb1-4248-8786-44a7b2ebdc8a.png" />     |

s77rt replied:
Hmm, I'm not sure I understand your question here.

If the report is a chat report (e.g. DM, Group, Room and Workspace chats) we will show the welcome message. Otherwise (IOU, expenseChat, Task) we won't.

Let me know if that answers your concern

s77rt replied:
> what was not working before and what does this change fix?

https://github.com/Expensify/App/pull/18775#discussion_r1194287708

Julesssss replied:
Sorry for the confusion all. I think I simply missed the `!` logic inversion when reviewing 😅

Julesssss replied:
I agree with this: 

> So instead of: expense || iou || task. Let's just do !chat.

s77rt replied:
Cool cool. Making this as resolved then :rocket:

s77rt committed:
Reduced section titles height

s77rt commented on src/styles/variables.js:
```diff
@@ -83,7 +83,7 @@ export default {
     optionRowHeight: 64,
     optionRowHeightCompact: 52,
-    optionsListSectionHeaderHeight: getValueUsingPixelRatio(54, 60),
+    optionsListSectionHeaderHeight: getValueUsingPixelRatio(32, 38),
```
Yes, discussed on [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1683999774534429?thread_ts=1683808646.650509&cid=C02NK2DQWUX).

s77rt commented:
@0xmiroslav Be sure to use the staging server (or enable staging switch)

therealsujitk commented:
I didn't realise this was already being worked on (#18991), btw from the screenshots there's some missing margin between the avatar and payee name no?

s77rt commented:
@therealsujitk Indeed. I guess we can add that margin but maybe we shouldn't? @shawnborton's call :grin:

Julesssss commented:
I just kicked off a CI run for generating the builds. Once this is complete let's aim for a final review. We're getting a bit too close to our conference deadline now.

Julesssss reviewed (dismissed):
Looking good IMO. This was originally a vague 'fix design issues' task, so I'd like to get it finished today. We can always create follow-up tasks for the remainder.

mountiny commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

mountiny commented:
@0xmiroslav that hsould be hopefully fixed soon, backend is currently being deployed

The job3 tst is failing tho I dont think we can merge it with that :( 
```
FAIL tests/ui/UnreadIndicatorsTest.js (52.362 s)
  ● Unread Indicators › Display bold in the LHN for unread chat and new line indicator above the chat message when we navigate to it

    expect(received).toBeTruthy()

    Received: null

      232 |                 const welcomeMessageHintText = Localize.translateLocal('accessibilityHints.chatWelcomeMessage');
      233 |                 const createdAction = screen.queryByLabelText(welcomeMessageHintText);
    > 234 |                 expect(createdAction).toBeTruthy();
          |                                       ^
      235 |                 const reportCommentsHintText = Localize.translateLocal('accessibilityHints.chatMessage');
      236 |                 const reportComments = screen.queryAllByLabelText(reportCommentsHintText);
      237 |                 expect(reportComments).toHaveLength(9);

      at toBeTruthy (tests/ui/UnreadIndicatorsTest.js:2[34](https://github.com/Expensify/App/actions/runs/4982738874/jobs/8918832553?pr=18775#step:6:35):[39](https://github.com/Expensify/App/actions/runs/4982738874/jobs/8918832553?pr=18775#step:6:40))
          at runMicrotasks (<anonymous>)
```

luacmartins commented:
@s77rt can you address @mountiny's comment above please?

luacmartins commented:
I don't think we should merge it with failing tests.

mountiny commented:
yeah we cannot merge this as then all PRs would be failing.

s77rt commented:
Sorry I thought the failing test was the old unread indicator one, it's another one. Fix is on the way.

s77rt committed:
Merge branch 'Expensify:main' into iou-header-polish

s77rt committed:
fix failing test

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/4995089873) to staging by https://github.com/luacmartins in version: 1.3.14-11 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4997415254) to production by https://github.com/yuwenmemon in version: 1.3.14-14 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/5273496004) to staging by https://github.com/AndrewGable in version: 1.3.28-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5293001429) to production by https://github.com/AndrewGable in version: 1.3.28-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

rushatgabhane commented on src/components/SubscriptAvatar.js:
```diff
@@ -63,23 +63,21 @@ const SubscriptAvatar = (props) => {
-            </View>
+            <Tooltip text={props.secondaryTooltip}>
+                <Avatar
+                    imageStyles={null}
```
Hi this PR caused https://github.com/Expensify/App/issues/19905

More details - https://github.com/Expensify/App/issues/19905#issuecomment-1593690817

s77rt replied:
Thanks @rushatgabhane We didn't test with a real image indeed. Nice catch!
